flows/analysis/build_defs.bzl

```diff
@@ -0,0 +1,120 @@
+
+    ctx.actions.write(
+        output = base_script,
+        content = "\n".join(tcl_commands + ["\n"]),
+    )
+
+    ctx.actions.run_shell(
+        inputs = [base_script, ctx.file.analysis_script],
+        outputs = [full_script],
+        command = "cat {base_script} {analysis_script} > {full_script}".format(
+            base_script = base_script.path,
+            analysis_script = ctx.file.analysis_script.path,
+            full_script = full_script.path,
+        ),
+    )
+
+    return assemble_openroad_step(
+        ctx,
+        ctx.attr.name,
+        full_script,
+        read_cells_runfiles,
+        inputs = ["netlist"] + ctx.attr.inputs,
+        outputs = ctx.attr.outputs,
+        constants = ["top"] + ctx.attr.outputs,
+    )
+
+analyze_netlist_step = rule(
+    implementation = _analyze_netlist_step_impl,
+    attrs = {
+        "_openroad": attr.label(
+            default = Label("@org_theopenroadproject//:openroad"),
+            executable = True,
+            cfg = "exec",
+        ),
+        "analysis_script": attr.label(
+            doc = "OpenROAD Tcl script implementing the analysis",
+            allow_single_file = [".tcl"],
+            mandatory = True,
+        ),
+        "standard_cells": attr.label(
+            doc = "Standard cells to use for analysis",
+            providers = [StandardCellInfo],
+            mandatory = True,
+        ),
+        "inputs": attr.string_list(
+            doc = "Additional file inputs (beyond netlist) required by this analysis step",
+        ),
+        "outputs": attr.string_list(
+            doc = "Names of file outputs generated by this analysis step",
+        ),
+        "constants": attr.string_list(
+            doc = "Additional string constants (beyond netlist top) required by this analysis step",
+        ),
+    },
+)
+
+def analyze_rtl_binary(
+        name,
+        standard_cells,
+        analysis_script,
+        inputs = [],
+        outputs = [],
+        constants = [],
+        yosys_script = "//flows/analysis:simple_synth.tcl"):
+    """Build an RTL analysis binary out of a flow that includes Yosys synthesis and OpenROAD analysis.
+
+    Args:
+      name: Name for the generated analysis binary.
+      standard_cells: StandardCellInfo provider for the cells used by Yosys and OpenROAD.
+      analysis_script: Tcl script used for OpenROAD analysis.
+      inputs: Names of logical inputs for the generated flow (beyond the rtl input for synthesis).
+      outputs: Names of logical outputs for the generated flow.
+      constants: Names of string constants required by the flow (beyond top).
+      yosys_script: Tcl script used for Yosys synthesis, defaults to //flows/analysis:simple_synth.tcl
+    """
+
+    yosys_step = name + "_yosys_step"
+    yosys_synth_file_step(name = yosys_step, standard_cells = standard_cells, synth_tcl = yosys_script)
+
+    analysis_step = name + "_analysis_step"
+    analyze_netlist_step(
+        name = analysis_step,
```

Comment:
Maybe state that these targets are present or make them hidden ("_"+name+....)? You can also change the visibility.